modules/ducktests/tests/ignitetest/services/utils/jvm_utils.py

```diff
@@ -0,0 +1,118 @@
+
+JVM_PARAMS_GC_G1 = "-XX:+UseG1GC -XX:MaxGCPauseMillis=100 " \
+                   "-XX:ConcGCThreads=$(((`nproc`/3)>1?(`nproc`/3):1)) " \
+                   "-XX:ParallelGCThreads=$(((`nproc`*3/4)>1?(`nproc`*3/4):1)) "
+
+JVM_PARAMS_GENERIC = "-server -da -XX:+DisableExplicitGC -XX:+AggressiveOpts -XX:+AlwaysPreTouch " \
+                     "-XX:+ParallelRefProcEnabled -XX:+DoEscapeAnalysis " \
+                     "-XX:+OptimizeStringConcat -XX:+UseStringDeduplication"
+
+
+def jvm_settings(heap_size="768M", gc_settings=JVM_PARAMS_GC_CMS, generic_params=JVM_PARAMS_GENERIC, gc_dump_path=None,
+                 oom_path=None, **kwargs):
+    """
+    Provides settings string for JVM process.
+    param as_list: Represent JVM params as list.
+    param as_map: Represent JVM params as dict.
+    param merge: JVM Params to merge. Adds new or rewrites default values. Can be dict, list or string.
+    """
+    gc_dump = "-XX:+UseGCLogFileRotation -XX:NumberOfGCLogFiles=5 -XX:GCLogFileSize=32M " \
+              "-XX:+PrintGCDateStamps -verbose:gc -XX:+PrintGCDetails -Xloggc:" + gc_dump_path \
+        if gc_dump_path else ""
+
+    out_of_mem_dump = "-XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=" + oom_path \
+        if oom_path else ""
+
+    as_string = f"-Xmx{heap_size} -Xms{heap_size} {gc_settings} {gc_dump} {out_of_mem_dump} {generic_params}".strip()
+
+    to_merge = kwargs.get("merge")
+
+    if to_merge or kwargs.get("as_map", False):
+        return jvm_settings_merge(as_string, to_merge if to_merge else {}, **kwargs)
+
+    if kwargs.get("as_list", False):
+        return as_string.split()
+
+    return as_string
+
+
+def jvm_settings_merge(src_settings, additionals, **kwargs):
+    """
+    Merges two JVM settings.
+    :param src_settings: base settings. Can be dict, list or string.
+    :param additionals: params to add to or overwrite in src_settings. Can be dict, list or string.
+    param as_list: If True, represents result as list.
+    param as_map: If True, represents result as dict.
+    :return merged JVM settings. By default as string.
+    """
+    as_map = _to_map(src_settings)
+
+    as_map.update(_to_map(additionals))
+
+    _remove_duplicates(as_map)
+
+    if kwargs.get("as_map", False):
+        return as_map
+
+    as_list = [e[0] + '=' + e[1] if len(e) > 0 and e[1] else e[0] for e in as_map.items()]
+
+    if kwargs.get("as_list", False):
+        return as_list
+
+    return ' '.join(as_list)
+
+
+def _to_map(params):
+    """"""
+    if isinstance(params, dict):
+        return params
+
+    if isinstance(params, list):
+        params = iter(params)
+    else:
+        params = str(params).split()
+
+    as_map = {}
+
+    for key_val in [entry.split(sep='=', maxsplit=1) for entry in params]:
+        as_map[str(key_val[0])] = str(key_val[1]) if len(key_val) > 1 else None
+
+    return as_map
+
+
+def _remove_duplicates(params: dict):
```

Comment:
As for me, this function is useless. It's result is non-determenistic. 
For example
```
>>> a = {"-Xmx":"123", "-xmx":"234", "-xMx":"345"}
>>>_remove_duplicates(a)
>>>a
{'-xMx': '345'}
```
And I suppose that this problem really is not a problem at all :). I can hardly believe, that someone add param
'-xMx'

Reply:
We have two params -Xmx in the test. Default and set for IgniteApplication for example. They both appear. This func. keeps last added.

Reply:
Python `dict` is hash table
```
>>> a = {'-Xmx': '4G', '-Xmx': '10G'}
>>> a
{'-Xmx': '10G'}
>>> 
```
So you should not convert to lowercase I suppose, current implementation is wrong, examples provided

Reply:
'Python dict is hash table

>>> a = {'-Xmx': '4G', '-Xmx': '10G'}
>>> a
{'-Xmx': '10G'}'

Yes, but in case of -Xmx and only keys works: 'Xmx4G' : None. Values are for '=' params

Reply:
Yes, I got it, but see an example -- _remove_duplicates can leave incorrect results and remove correct.
So you need add additional conditions.

That's why I asked you to add unit tests.

Reply:
I suppose not =lowering is enough. Fixed.

Reply:
Ok, but you can do it in one iteration.

```
 duplicates = {"-Xmx": False,  "-Xms": False}

for param_key in list(params.keys()):
     for dup_key in duplicates.keys():
           if param_key.startswith(dup_key):
               if duplicates[dup_key]:
                    del params[param_key]
               else:
                   duplicates[dup_key] = True
              
```

Reply:
Fixed

Reply:
Why only Xmx and Xms are on the agenda?
How about -da and -ea? _remove_duplicates will do nothing in this case.

Also, AFAIU we have generic params set, and we may merge it with something. The strategy seems odd to me. 

As for me, we should append params from generic's set to user's set in they are missed in user's, but this also does not solve the -da/-ea problem.

Reply:
Because memory setting is critical and conflicts. We use default values and reset then. But this method is to resolve all options conflicts. Not intended to resolve them all in this ticket.

Reply:
We have default params and we wont to rewrite them sometimes.